astropy/units/core.py

```diff
@@ -2565,6 +2563,26 @@ def _add_prefixes(u, excludes=[], namespace=None, prefixes=False):
             )
 
 
+@overload
+def def_unit(
+    s: str | list[str],
+    represents: UnitBase,
+    doc: str | None = ...,
```

Comment:
I'm not confident that elliding default values is the right thing to do. In general, I find that `mypy` is much easier to satisfy with `overload`s, but also less generally useful than, say, `pyright`, which flags the following example

```python
from typing import overload, Any

@overload
def foo(
    a: bool | None = ...,
    b: None = ...,
) -> None: ...
@overload
def foo(
    a: bool | None = ...,
    b: bool = ...,
) -> bool: ...
def foo(a: bool | None = None, b: bool | None = None) -> Any: ...
```

with
```
/private/tmp/test/t.py
  /private/tmp/test/t.py:9:5 - error: Overload 1 for "foo" overlaps overload 2 and returns an incompatible type (reportOverlappingOverload)
1 error, 0 warnings, 0 informations 
```

Reply:
OK